### PR TITLE
rocksdb-in-tikv bench tool

### PR DESCRIPTION
**Pull request by innerr**
My first rust project, any help will be appreciated

### CODE REVIEW

File: rocksdb-in-tikv/src/sim/val.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (siddontang) [inline rocksdb-in-tikv/src/sim/val.rs]: use 2017

File: rocksdb-in-tikv/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+fn run() -> Result<usize, String> {
+    let app = App::new("Rocksdb in TiKV")
+        .author("PingCAP")
+        .about("Benchmark of rocksdb in the sim-tikv-env")
+        .arg(Arg::with_name("db")
+            .short("db")
+            .takes_value(true)
+            .help("rocksdb path")
+            .required(true))
+        .arg(Arg::with_name("cfg1")
```
Comment (siddontang) [inline rocksdb-in-tikv/src/main.rs]: use meaningful name, not 1 and 2

File: rocksdb-in-tikv/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "rocksdb-in-tikv"
+version = "0.1.0"
+authors = ["Liu Cong <innerr@gmail.com>"]
```
Comment (zhangjinpeng1987) [inline rocksdb-in-tikv/Cargo.toml]: Please use pingcap email.

File: rocksdb-in-tikv/cases/cf_commit.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[rocksdb]
```
Comment (siddontang) [inline rocksdb-in-tikv/cases/cf_commit.toml]: we don't have commit and data cf.

File: rocksdb-in-tikv/Makefile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+build:
+	RUST_BACKTRACE=1 cargo build --release --features "${ENABLE_FEATURES}"
+	@mkdir -p ${BIN_PATH}
+	cp -f ${CARGO_TARGET_DIR}/release/rocksdb-in-tikv ${BIN_PATH}/
+
+static_release:
+	ROCKSDB_SYS_STATIC=1 ROCKSDB_SYS_PORTABLE=1 ROCKSDB_SYS_SSE=1  make release
+
+static_unportable_release:
+	ROCKSDB_SYS_STATIC=1 ROCKSDB_SYS_SSE=1  make release
```
Comment (siddontang) [inline rocksdb-in-tikv/Makefile]: no release label

File: rocksdb-in-tikv/cases/base.toml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+max-bytes-for-level-base = "512MB"
+target-file-size-base = "32MB"
+block-cache-size = "1GB"
+cache-index-and-filter-blocks = true
+
+block-based-bloom-filter = false
+bloom-filter-bits-per-key = 10
+
+compression-per-level = "no:no:lz4:lz4:lz4:lz4:lz4"
+#compression-per-level = "no:no:lz4:lz4:lz4:zstd:zstd"
```
Comment (zhangjinpeng1987) [inline rocksdb-in-tikv/cases/base.toml]: Please remove this line.

File: rocksdb-in-tikv/src/env/check.rs [hunk 7]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (siddontang) [inline rocksdb-in-tikv/src/env/check.rs]: is it for the test, so I prefer setting these directly in the test script, no need to check here.

File: rocksdb-in-tikv/cases/cf_commit.toml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+min-write-buffer-number-to-merge = 1
+max-bytes-for-level-base = "512MB"
+target-file-size-base = "32MB"
+cache-index-and-filter-blocks = true
+level0-file-num-compaction-trigger = 4
+level0-slowdown-writes-trigger = 20
+level0-stop-writes-trigger = 36
+cache-index-and-filter-blocks = true
+
+# in normal cases should tune to 10%-30% system's total memory.
```
Comment (zhangjinpeng1987) [inline rocksdb-in-tikv/cases/cf_commit.toml]: please remove this line.

File: rocksdb-in-tikv/src/env/dbcfg.rs [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use process;
```
Comment (siddontang) [inline rocksdb-in-tikv/src/env/dbcfg.rs]: why not import tikv library directly?

File: rocksdb-in-tikv/cases/base.toml [hunk 10]
```diff
@@ -0,0 +1,42 @@
+[rocksdb]
```
Comment (siddontang) [inline rocksdb-in-tikv/cases/base.toml]: IMO, I suggest putting all configs into a file, it seems more easy to manage and parse.

File: rocksdb-in-tikv/src/env/dbcfg.rs (hunk 9 above)
Comment (innerr) [inline rocksdb-in-tikv/src/env/dbcfg.rs]: Tried, too big and build slow. I perfer remove default_values so import tikv not a good choice.

File: rocksdb-in-tikv/cases/base.toml (hunk 10 above)
Comment (innerr) [inline rocksdb-in-tikv/cases/base.toml]: Depends on how many rocksdb args we need to care (in the future), if it grows to >50  I perfer the base+tuning file mode.
If it remains <20, one file would be good.

File: rocksdb-in-tikv/src/env/check.rs (hunk 7 above)
Comment (innerr) [inline rocksdb-in-tikv/src/env/check.rs]: Good point.

File: rocksdb-in-tikv/cases/cf_commit.toml (hunk 4 above)
Comment (innerr) [inline rocksdb-in-tikv/cases/cf_commit.toml]: It's "default" and "write" cf, I find the names are a lit too common.

**Comment by innerr** [comment]
Address comments. PTAL

**Comment by innerr** [comment]
When an arg is not defined in the toml file, It's hard to tale it would be the TiKV-default-value or the RocksDB-default-value.
So I prefer using "base file (as default values, act like rocksdb-default if one arg not set)" + "tuning file", to reduce the ambiguous meaning.

File: rocksdb-in-tikv/cases/cf_commit.toml (hunk 4 above)
Comment (siddontang) [inline rocksdb-in-tikv/cases/cf_commit.toml]: should keep the same name with TiKV

File: rocksdb-in-tikv/src/env/dbcfg.rs (hunk 9 above)
Comment (siddontang) [inline rocksdb-in-tikv/src/env/dbcfg.rs]: yes, I think we can put all fields in config file directly.

File: rocksdb-in-tikv/cases/cf_commit.toml (hunk 4 above)
Comment (innerr) [inline rocksdb-in-tikv/cases/cf_commit.toml]: Will do.

**Pull request merged**